### PR TITLE
Adds qdel defines

### PR DESCRIPTION
**Pull request by SECBATON-GRIFFON**
> QDEL_NULL() is one of TG's actually reasonable macros and we should steal it

[qol]

## What this does
[port]s three defines from TG, QDEL_NULL, QDEL_LIST and QDEL_LIST_ASSOC, the first handles any usage of ```qdel(item); item = null``` the second deletes all items in a list and calls Cut(), the third does the same for associative lists.

## Why it's good
cuts down a looooot of code

### CODE REVIEW

File: code/ATMOSPHERICS/components/binary_devices/binary_atmos_base.dm [hunk 1]
```diff
@@ -215,8 +215,8 @@
 
 	else if(reference==node2)
 		if(network2)
-			qdel(network2)
-		node2 = null
+			qdel(network1)
+		node1 = null
```
Comment (Exxion) [inline code/ATMOSPHERICS/components/binary_devices/binary_atmos_base.dm]: ?

File: code/ATMOSPHERICS/components/trinary_devices/trinary_base.dm [hunk 2]
```diff
@@ -225,12 +225,12 @@
...
 		if(network2)
 			qdel(network2)
-		node2 = null
+		node1 = null
 
 	else if(reference==node3)
 		if(network3)
 			qdel(network3)
-		node3 = null
+		node1 = null
```
Comment (Exxion) [inline code/ATMOSPHERICS/components/trinary_devices/trinary_base.dm]: ?

File: code/ATMOSPHERICS/components/binary_devices/binary_atmos_base.dm (hunk 1 above)
Comment (SECBATON-GRIFFON) [reply code/ATMOSPHERICS/components/binary_devices/binary_atmos_base.dm]: must've missed one

File: code/ATMOSPHERICS/components/trinary_devices/trinary_base.dm (hunk 2 above)
Comment (SECBATON-GRIFFON) [reply code/ATMOSPHERICS/components/trinary_devices/trinary_base.dm]: and another

**Comment by SECBATON-GRIFFON** [comment]
after qdel()ing everything in lists, what's better to write? list = null, list = list() or list.Cut()? the first shows up the most but the last is in the define i found.

**Comment by Exxion** [comment]
Depends on the situation. Sometimes the first, sometimes the third. The second and third are (almost) functionally equivalent but the third is slightly faster. Both of them are a waste of operations when the first could be used, though (such as in `Destroy()`).
In general, you'd want the first when you're done with the list and the third when you're emptying it but intend to use it again

**Comment by SECBATON-GRIFFON** [comment]
> Depends on the situation. Sometimes the first, sometimes the third. The second and third are (almost) functionally equivalent but the third is slightly faster. Both of them are a waste of operations when the first could be used, though (such as in `Destroy()`). In general, you'd want the first when you're done with the list and the third when you're emptying it but intend to use it again

added both as defines now, put the = null ones in action

**Comment by adacovsk** [comment]
I heard from @DamianX that it's faster to check if a list is null before you .cut()

**Comment by Exxion** [comment]
You mean empty? Not faster if it's not empty

**Pull request merged**